### DISCUSSION TITLE
How to import  external library into boot file

### DISCUSSION THREAD

**Discussion by vicman1200** [quasarframework/quasar#9913] [General - Components / Directives / etc] [answered]
Hi!

I'am trying to add a [Google OAuth external library](https://www.npmjs.com/package/vue-google-oauth2) as boot file, doc say to add some code into main.js:

```
//main.js
import GAuth from 'vue-google-oauth2'
const gauthOption = {
  clientId: 'CLIENT_ID.apps.googleusercontent.com',
  scope: 'profile email',
  prompt: 'select_account'
}
Vue.use(GAuth, gauthOption)
```

 

and this is my code in boot file in Quasar project:

src/boot/googleOAuth2.js

```
import { boot } from 'quasar/wrappers'
import GAuth from 'vue-google-oauth2'
const gauthOption = {
  clientId: 'CLIENT_ID.apps.googleusercontent.com',
  scope: 'profile email',
  prompt: 'select_account'
}

export default boot(({ app }) => {
  // something to do 
      app.use(GAuth, gauthOption)
  })
```

I'm getting next error:

`[Quasar] boot error: TypeError: Object.defineProperties called on non-object`

 What is missing? 😕

Thanks in advance!

**Comment by rstoenescu** [comment, marked as answer]
From the description that you provided I can infer that you are using Quasar v2 (so Vue 3) and `vue-google-oauth2` is meant for Vue 2 and NOT for Vue 3. You can open a ticket on vue-google-oauth2 repo to request compatibility with Vue 3.

**Reply by vicman1200** [reply]
Your right!

Thank you! 🍺🍺🍺
